format.go

```diff
@@ -0,0 +1,232 @@
+package logged
+
+import (
+	"fmt"
+	"time"
+)
+
+const (
+	LevelKey   = "lvl"
+	MessageKey = "msg"
+
+	timeFormat = "2006-01-02T15:04:05-0700" // ISO8601 format
+)
+
+// Formatter represents a log message formatter.
+type Formatter interface {
+	// Format formats a log message.
+	Format(msg string, lvl Level, ctx []interface{}) []byte
+}
+
+// FormatterFunc is a function formatter.
+type FormatterFunc func(msg string, lvl Level, ctx []interface{}) []byte
+
+// Format formats a log message.
+func (f FormatterFunc) Format(msg string, lvl Level, ctx []interface{}) []byte {
+	return f(msg, lvl, ctx)
+}
+
+var jsonPool = NewPool(512)
+
+// JsonFormatter formats a log line in json format.
+func JsonFormat() Formatter {
+	return FormatterFunc(func(msg string, lvl Level, ctx []interface{}) []byte {
+		buf := jsonPool.Get()
+
+		// Append initial keys to the buffer
+		buf.WriteByte('{')
+		buf.WriteString(`"` + LevelKey + `":"` + lvl.String() + `",`)
+		buf.WriteString(`"` + MessageKey + `":`)
+		quoteString(buf, msg)
+
+		// Append ctx to the buffer
+		for i := 0; i < len(ctx); i += 2 {
+			buf.WriteByte(',')
+
+			k, ok := ctx[i].(string)
+			if !ok {
+				buf.WriteString(`"` + errorKey + `"`)
+				buf.WriteByte(':')
+				formatJsonValue(buf, ctx[i])
+				continue
+			}
+
+			buf.WriteString(`"` + k + `"`)
+			buf.WriteByte(':')
+			formatJsonValue(buf, ctx[i+1])
+		}
+
+		buf.WriteString("}\n")
+
+		jsonPool.Put(buf)
+		return buf.Bytes()
+	})
+}
+
+// formatJsonValue formats a value, adding it to the Buffer.
+func formatJsonValue(buf *Buffer, value interface{}) {
+	if value == nil {
+		buf.WriteString("null")
+		return
+	}
+
+	switch v := value.(type) {
+	case time.Time:
+		buf.WriteByte('"')
+		buf.AppendTime(v, timeFormat)
+		buf.WriteByte('"')
+	case bool:
+		buf.AppendBool(v)
+	case float32:
+		buf.AppendFloat(float64(v), 'g', -1, 64)
+	case float64:
+		buf.AppendFloat(v, 'g', -1, 64)
+	case int:
+		buf.AppendInt(int64(v))
+	case int8:
+		buf.AppendInt(int64(v))
+	case int16:
+		buf.AppendInt(int64(v))
+	case int32:
+		buf.AppendInt(int64(v))
+	case int64:
+		buf.AppendInt(v)
```

Comment:
All the int cases could be merged into 1. Why not do it and avoid repetition?

Reply:
Same for uints and floats.

Reply:
How is that?